### PR TITLE
Fix some Flax models' `hidden_states`

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Fix some Flax models where the last element in `hidden_states` is different between PT/Flax version.

## More context

Some models have 

```
last_hidden_state = self.layer_norm(last_hidden_state)
```

In Pytorch version, the returned `hidden_states` have this `last_hidden_state` (after layer norm) as the last element.
In Flax version, the last element of the returned `hidden_states` is the one before the layer norm.

This PR fixes this inconsistency (by using the PyTorch logic).

### CODE REVIEW

File: src/transformers/models/blenderbot/modeling_flax_blenderbot.py [hunk 1]
```diff
@@ -782,12 +791,21 @@ def __call__(
...
+        if output_hidden_states:
+            hidden_states = outputs[1]
+            hidden_states = hidden_states[:-1] + (last_hidden_states,)
+
         if not return_dict:
-            return (last_hidden_states,) + outputs[1:]
+            if not output_hidden_states:
+                return (last_hidden_states,) + outputs[1:]
+            else:
+                return (last_hidden_states, hidden_states) + outputs[2:]
```
Comment (patil-suraj) [inline src/transformers/models/blenderbot/modeling_flax_blenderbot.py:804]: same comment as above

File: src/transformers/models/mbart/modeling_flax_mbart.py [hunk 2]
```diff
@@ -768,12 +768,21 @@ def __call__(
...
+        if output_hidden_states:
+            hidden_states = outputs[1]
+            hidden_states = hidden_states[:-1] + (last_hidden_states,)
+
         if not return_dict:
-            return (last_hidden_states,) + outputs[1:]
+            if not output_hidden_states:
+                return (last_hidden_states,) + outputs[1:]
+            else:
+                return (last_hidden_states, hidden_states) + outputs[2:]
```
Comment (patil-suraj) [inline src/transformers/models/mbart/modeling_flax_mbart.py:781]: same comment as above

**Review by patil-suraj — APPROVED** [review]
Great catch! Thanks a lot for fixing this.

Just left a nit :)

File: src/transformers/models/blenderbot/modeling_flax_blenderbot.py (hunk 1 above)
Comment (ydshieh) [reply src/transformers/models/blenderbot/modeling_flax_blenderbot.py:802]: Hi, is this better

https://github.com/huggingface/transformers/blob/415eb3c0ff9cfd68f3ba25122370324220f01b87/src/transformers/models/blenderbot/modeling_flax_blenderbot.py#L720-L722

(when it returns tuple + when it needs extra processing as in this PR, I always have trouble to make it cleaner. Things get much easier if the internal components return dict or named tuple, and only change the format at the top level components for the users - but I don't think we are going to do so, at least not soon)

Comment (patil-suraj) [reply src/transformers/models/blenderbot/modeling_flax_blenderbot.py:802]: This looks good!

> Things get much easier if the internal components return dict or named tuple, and only change the format at the top level components for the users

This is a good idea, I think we can change the internal modules to only return either dict or Tuple

Comment (ydshieh) [reply src/transformers/models/blenderbot/modeling_flax_blenderbot.py:802]: Thanks for the feedback :-) @patil-suraj . I will apply the same change to other places to finish this PR.
About changing  internal components (in general), let's have a discussion later with other members.

**Review by patrickvonplaten — APPROVED** [review]
Thanks!

**Pull request merged**